core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -67,12 +67,16 @@ class AdminClient(val time: Time,
     throw new RuntimeException(s"Request $api failed on brokers $bootstrapBrokers")
   }
 
-  private def findCoordinator(groupId: String): Node = {
+  private def findCoordinator(groupId: String): Option[Node] = {
     val request = new GroupCoordinatorRequest(groupId)
     val responseBody = sendAnyNode(ApiKeys.GROUP_COORDINATOR, request)
     val response = new GroupCoordinatorResponse(responseBody)
-    Errors.forCode(response.errorCode()).maybeThrow()
-    response.node()
+    Errors.forCode(response.errorCode()) match {
+      case Errors.GROUP_COORDINATOR_NOT_AVAILABLE => None
+      case error: Errors =>
```

Comment:
No type annotation should be used here (i.e. `: Errors`). Why? Without it, it will catch everything. And if the type is not the one you expect, the compiler will complain when you call one of its methods. With the type annotation, it will only catch `Errors`. If you pass something else, it will throw a `MatchError`.